src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas3/Cas3AssessmentService.kt

```diff
@@ -36,22 +37,52 @@ class Cas3AssessmentService(
       return CasResult.GeneralValidationError("Cannot update both dates")
     }
 
-    updateAssessment.releaseDate?.apply {
-      if (this.isAfter(assessment.currentAccommodationRequiredFromDate())) return notAfterValidationResult(assessment.currentAccommodationRequiredFromDate())
-      assessment.releaseDate = this
+    updateAssessment.releaseDate?.let { newReleaseDate ->
+      val currentAccommodationReqDate = assessment.currentAccommodationRequiredFromDate()
+
+      if (newReleaseDate.isAfter(currentAccommodationReqDate)) {
+        return notAfterValidationResult(currentAccommodationReqDate)
+      }
+
+      val domainEvent =
+        domainEventBuilder.buildAssessmentUpdatedDomainEvent(
+          assessment = assessment,
+          fieldName = "releaseDate",
+          updatedFrom = assessment.currentReleaseDate().toString(),
+          updatedTo = newReleaseDate.toString(),
```

Comment:
in CAS1 we use the LocalDate.toUiFormat() extension function in Dates.kt whenever converting dates to strings. Is LocalDate.toString() producing dates that match how you present dates elsewhere on the UI? (i'm assuming this will be shown in a timeline somewhere?)